application-admintools-default/src/main/java/com/xwiki/admintools/internal/files/resources/XWikiPropertiesFileDataResource.java

```diff
@@ -0,0 +1,130 @@
+import java.util.zip.ZipOutputStream;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+
+import com.xwiki.admintools.configuration.AdminToolsConfiguration;
+import com.xwiki.admintools.download.DataResource;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+
+/**
+ * {@link DataResource} implementation for accessing the xwiki.properties file.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named(XWikiPropertiesFileDataResource.HINT)
+@Singleton
+public class XWikiPropertiesFileDataResource implements DataResource
+{
+    /**
+     * Component identifier.
+     */
+    public static final String HINT = "xwikiProperties";
+
+    private static final String XWIKI_PROPERTIES = "xwiki.properties";
+
+    private static final String ERROR_SOURCE = " Root cause is: [{}]";
+
+    @Inject
+    @Named("default")
+    private AdminToolsConfiguration adminToolsConfig;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public void addZipEntry(ZipOutputStream zipOutputStream, Map<String, String> filters) throws IOException
+    {
+        addZipEntry(zipOutputStream);
+    }
+
+    @Override
+    public byte[] getByteData(String input) throws Exception
+    {
+        try {
+            List<String> excludedLinesHints = adminToolsConfig.getExcludedLines();
+            String filePath = currentServer.getCurrentServer().getXwikiCfgFolderPath() + XWIKI_PROPERTIES;
+            File inputFile = new File(filePath);
+            try (BufferedReader reader = new BufferedReader(new FileReader(inputFile))) {
+                StringBuilder stringBuilder = new StringBuilder();
+                String currentLine;
+
+                // Read line by line and do not add it if it contains sensitive info.
+                while ((currentLine = reader.readLine()) != null) {
+                    String trimmedLine = currentLine.trim();
+                    if (excludedLinesHints.stream().anyMatch(trimmedLine::contains)) {
+                        continue;
+                    }
+                    stringBuilder.append(currentLine).append(System.getProperty("line.separator"));
+                }
+                return stringBuilder.toString().getBytes();
+            }
+        } catch (IOException exception) {
+            String errMessage = String.format("Could not find %s file.", XWIKI_PROPERTIES);
+            logger.warn(errMessage + ERROR_SOURCE, ExceptionUtils.getRootCauseMessage(exception));
+            throw new IOException(errMessage, exception);
+        } catch (Exception e) {
+            String errMessage = String.format("Failed to get content of %s.", XWIKI_PROPERTIES);
+            logger.warn(errMessage + ERROR_SOURCE, ExceptionUtils.getRootCauseMessage(e));
```

Comment:
I think you can remove some `logger` calls in here. Let's look at how this method is used in this flow:
`DefaultAdminToolsResource#getFile -> ImportantFilesManager#getFile -> XWikiPropertiesFileDataResource#getByteData`
Right now you are logging a message from getByteData and then logging something pretty similar inside the `DefaultAdminToolsResource#getFile` , since ExceptionUtils.getRootCauseMessage(e) gives you the same root cause:
```
"Failed to get content of XWikiProperties. Root cause is: [Exception cause 1]"
"Could not find file from DataResource[XWikiProperties]. Root cause: [Exception cause 1]"
```
So you should log just once, at the end of you exception handling, as long as you make sure to use the correct root cause. You can also access the exception message, if there is something specific to mention

Reply:
Please check your other classes as well